### PR TITLE
[REF] Convert Elavon Payment Processor to use Guzzle and ensure that …

### PR DESCRIPTION
**Pull request by seamuslee001**
…we are using POST requests not GET requests to Elavon end point

Overview
----------------------------------------
This updates the Elavon Payment Processor to use Guzzle and also use POST requests rather than Get

Before
----------------------------------------
Curl used and GET requests used

After
----------------------------------------
Guzzle used and POST requests used

ping @eileenmcnaughton @monishdeb @JoeMurray

### CODE REVIEW

File: CRM/Core/Payment/Elavon.php [hunk 1]
```diff
@@ -142,71 +161,25 @@ public function doPayment(&$params, $component = 'contribute') {
...
-      throw new PaymentProcessorException('Error: Connection to payment gateway failed - no data returned.', 9006);
-    }
+    $responseData = $this->getGuzzleClient()->post($chHost, [
+      'curl' => [
+        CURLOPT_RETURNTRANSFER => TRUE,
+        CURLOPT_TIMEOUT => 36000,
+        CURLOPT_SSL_VERIFYHOST => Civi::settings()->get('verifySSL') ? 2 : 0,
+        CURLOPT_SSL_VERIFYPEER => Civi::settings()->get('verifySSL'),
+      ],
+    ])->getBody();
```
Comment (monishdeb) [inline CRM/Core/Payment/Elavon.php:171]: @seamuslee001 is it required to set the `CURLOPT_FOLLOWLOCATION` if as per L160 above?

**Comment by eileenmcnaughton** [comment]
I don't suppose we can move it into an extension while we are touching it....

Are you able to test it with any test creds? Do we know if it even works? If it were an extension we could not support it for php 8....

**Comment by seamuslee001** [comment]
@eileenmcnaughton I would rather not, I only put this up because I had to do this for a client today to fix their processing

File: CRM/Core/Payment/Elavon.php (hunk 1 above)
Comment (seamuslee001) [reply CRM/Core/Payment/Elavon.php:175]: ok have updated to it @monishdeb

**Comment by eileenmcnaughton** [comment]
Merging this based on discussion with @seamuslee001 - key points
- their client already has this patch in prodcution
- - Seamus will follow up with an extension conversion

**Pull request merged**